local/network.go

```diff
@@ -342,12 +345,15 @@ func (ln *localNetwork) addNode(nodeConfig node.Config) (node.Node, error) {
 		}
 	} else {
 		// Get a free port for the API port
-		l, err := net.Listen("tcp", ":0")
-		if err != nil {
-			return nil, fmt.Errorf("could not get free port: %w", err)
+		for {
+			r := rand.Intn(maxPort-minPort) + minPort
+			l, err := net.Listen("tcp", fmt.Sprintf(":%d", r))
+			if err == nil {
+				apiPort = uint16(l.Addr().(*net.TCPAddr).Port)
+				_ = l.Close()
+				break
```

Comment:
1. Let's add some comments explaining what we're doing
2. As you suggested on Slack, let's use a context with a timeout and a select statement to make sure this doesn't loop forever